palette/colormap.go

```diff
@@ -0,0 +1,150 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package palette
+
+import (
+	"fmt"
+	"image/color"
+	"math"
+
+	"github.com/gonum/floats"
+)
+
+// FromPalette creates a ColorMap that maps values to colors based
+// on RGB interpolation among the colors in p.
+func FromPalette(p Palette) ColorMap {
+	scalars := make([]float64, len(p.Colors()))
+	return &colorMap{
+		colors:  p.Colors(),
+		scalars: floats.Span(scalars, 0, 1),
+		alpha:   1,
+	}
+}
+
+type colorMap struct {
```

Comment:
To minimize the "weight of things". With the current ColorMap interface it doesn't contain anything the user needs to interact with directly. Under proposal #352 it would need to be exported though.